site/src/esl-popup/esl-d-popup-game.ts

```diff
@@ -0,0 +1,91 @@
+import {ESLBaseElement} from '@exadel/esl/modules/esl-base-element/core';
+import {attr, boolAttr, decorate, listen, memoize} from '@exadel/esl/modules/esl-utils/decorators';
+import {rafDecorator} from '@exadel/esl/modules//esl-utils/async';
+import {ESLResizeObserverTarget} from '@exadel/esl/modules/esl-utils/dom/events';
+import {ESLTraversingQuery} from '@exadel/esl/modules/esl-traversing-query/core';
+
+/** Fun component like a game for checking popup positioning on edge cases */
+export class ESLDemoPopupGame extends ESLBaseElement {
+  static override is = 'esl-d-popup-game';
+
+  /** Trigger selector */
+  @attr() public trigger: string;
+  /** Marker shows that element is in dragging mode */
+  @boolAttr() public dragging = false;
+
+  @memoize()
+  get $trigger(): HTMLElement | undefined {
+    if (this.trigger) return ESLTraversingQuery.first(this.trigger, this) as HTMLElement | undefined;
+  }
+
+  protected update(): void {
+    if (!this.$trigger) return;
+    if (this.dragging) this._onDragStop();
+    this.$$off(this._onDragStart);
+    this.$$on(this._onDragStart);
```

Comment:
What the purpose for resubscription of the unconditionl, self-applied (target can not be changed) listener ?

Reply:
This is redundant code. It was the first attempt to construct a component. Probably I missed it to remove.
Thanks a lot!